ledger/participant-integration-api/src/main/resources/db/migration/postgres/V50__Append_only_schema.sql

```diff
@@ -121,52 +121,541 @@ CREATE TABLE participant_events_divulgence             PARTITION OF participant_
 
 -- lookup by contract id
-CREATE INDEX participant_events_create_contract_id_idx ON participant_events_create USING hash (contract_id);
 CREATE INDEX participant_events_consuming_exercise_contract_id_idx ON participant_events_consuming_exercise USING hash (contract_id);
 
--- lookup divulgance events, in order of ingestion
-CREATE INDEX participant_events_divulgence_contract_id_idx ON participant_events_divulgence USING btree (contract_id, event_sequential_id);
 
--- lookup by contract_key
-CREATE INDEX participant_events_create_create_key_hash_idx ON participant_events_create USING btree (create_key_hash, event_sequential_id);
+
+---------------------------------------------------------------------------------------------------
+-- Events table: non-consuming exercise
+---------------------------------------------------------------------------------------------------
+CREATE TABLE participant_events_non_consuming_exercise (
+    -- * kinds of events
+    event_kind smallint NOT NULL  CONSTRAINT non_consuming_exercise_event_kind CHECK (event_kind = 25),
+    -- Numbers allocated to leave some space for future additions.
+    -- 0:  divulgence event
+    -- 10: create event
+    -- 20: consuming exercise event
+    -- 25: non-consuming exercise event
+
+    -- * event identification
+    event_sequential_id bigserial NOT NULL, -- TODO append-only: temporarily readding bigserial for original write paths
+    -- NOTE: this must be assigned sequentially by the indexer such that
+    -- for all events ev1, ev2 it holds that '(ev1.offset < ev2.offset) <=> (ev1.event_sequential_id < ev2.event_sequential_id)
+
+    event_offset bytea,
+
+    -- * transaction metadata
+    transaction_id text,
+    ledger_effective_time timestamp without time zone,
+    command_id text,
+    workflow_id text,
+    application_id text,
+    submitters text[],
+
+    -- * event metadata
+    node_index integer,
+    event_id text,        -- string representation of (transaction_id, node_index)
+
+    -- * shared event information
+    contract_id text NOT NULL,
+    template_id text,     -- TODO @simon@ with the new divulgance model supporting public pkv implementations: we need this to enable NULL-s. Do we need to make involved indexes partial?
+    flat_event_witnesses text[] DEFAULT '{}'::text[] NOT NULL,       -- stakeholders of create events and consuming exercise events
+    tree_event_witnesses text[] DEFAULT '{}'::text[] NOT NULL,       -- informees for create, exercise, and divulgance events
+
+    -- * divulgence and create events
+    -- [N/A for this event kind]: create_argument bytea,
+
+    -- * create events only
+    -- [N/A for this event kind]: create_signatories text[],
+    -- [N/A for this event kind]: create_observers text[],
+    -- [N/A for this event kind]: create_agreement_text text,
+    -- [N/A for this event kind]: create_key_value bytea,
+    -- [N/A for this event kind]: create_key_hash bytea,
```

Comment:
@rautenrieth-da : what's your view on removing the comments here and adding the comments where necessary on the view below?

Reply:
The comments are only here to simplify review. I will remove them all in a separate commit before undrafting the PR.